### PR TITLE
Feat/127/crypto conditions ilp bigchain integration hashlock

### PR DESCRIPTION
**Pull request by diminator**
This PR builds on top of #174 and provides anonymous hashlocked conditions and fulfillments

Should be reviewed after #174. Once #174 is merged the diff for this PR will be much <sub>smaller</sub>

### CODE REVIEW

File: bigchaindb/util.py [hunk 1]
```diff
@@ -247,20 +256,24 @@ def create_tx(current_owners, new_owners, inputs, operation, payload=None):
     # handle outputs
     conditions = []
     for fulfillment in fulfillments:
+        condition = None
```
Comment (r-marques) [inline bigchaindb/util.py]: Maybe cleaner to create a new `elif` branch.

```python
# threshold condition
if len(new_owners) > 1:
    ...
# simple signature condition
elif len(new_owners) == 1:
    ...
# to be added later (hashlock conditions)
elif not new_owners:
    condition = None
```

File: docs/source/python-server-api-examples.md [hunk 2]
```diff
@@ -746,3 +746,149 @@ threshold_tx_transfer
...
+    'condition': {
+        'details': json.loads(first_tx_condition.serialize_json()),
+        'uri': first_tx_condition.condition.serialize_uri()
+    },
+    'cid': 0,
+    'new_owners': None
+})
+
+# Conditions have been updated, so hash needs updating
+hashlock_tx['id'] = util.get_hash_data(hashlock_tx)
```
Comment (r-marques) [inline docs/source/python-server-api-examples.md]: `util.get_hash_data(hashlock_tx['transaction'])`

File: docs/source/python-server-api-examples.md [hunk 3]
```diff
@@ -746,3 +746,149 @@ threshold_tx_transfer
...
+    "assignee":"FmLm6MxCABc8TsiZKdeYaZKo5yZWMM6Vty7Q1B6EgcP2",
+    "id":"604c520244b7ff63604527baf269e0cbfb887122f503703120fd347d6b99a237",
+    "transaction":{
+        "conditions":[
+            {
+                "cid":0,
+                "condition":{
+                    "details":{
+                        "bitmask":3,
+                        "preimage":"much secret! wow!",
```
Comment (r-marques) [inline docs/source/python-server-api-examples.md]: the preimage is published with the condition?

File: docs/source/python-server-api-examples.md (hunk 2 above)
Comment (r-marques) [inline docs/source/python-server-api-examples.md]: forget this. `get_hash_data` already checks for this

File: docs/source/python-server-api-examples.md (hunk 3 above)
Comment (diminator) [inline docs/source/python-server-api-examples.md]: hahaha, Yeah, I should do something about that

File: tests/doc/run_doc_python_server_api_examples.py [hunk 4]
```diff
@@ -233,4 +233,61 @@
...
 
+"""
+Hashlocked Conditions
+"""
+
+# Create a hash-locked asset without any new_owners
+hashlock_tx = b.create_transaction(b.me, None, None, 'CREATE')
+
+# Define a secret that will be hashed - fulfillments need to guess the secret
+secret = b'much secret! wow!'
```
Comment (vrde) [inline tests/doc/run_doc_python_server_api_examples.py]: image

**Comment by vrde** [comment]
:+1:

**Pull request merged**